drivers/timer/arcv2_timer0.c

```diff
@@ -351,21 +351,9 @@ u32_t z_timer_cycle_get_32(void)
  */
 void sys_clock_disable(void)
 {
-	unsigned int key;  /* interrupt lock level */
-	u32_t control; /* timer control register value */
-
-	key = irq_lock();
-
-	/* disable interrupt generation */
-
-	control = timer0_control_register_get();
-	timer0_control_register_set(control & ~_ARC_V2_TMR_CTRL_IE);
-
-	irq_unlock(key);
-
-	/* disable interrupt in the interrupt controller */
-
-	irq_disable(IRQ_TIMER0);
+	/* This function is only called in sys_reboot,
```

Comment:
There's a __weak definition in sys_clock_init.c already, why not simply remove completely?